src/components/linter.ts

```diff
@@ -1,248 +1,50 @@
 import * as vscode from 'vscode'
-import * as path from 'path'
-import * as fs from 'fs'
-import * as os from 'os'
-import {ChildProcessWithoutNullStreams, spawn} from 'child_process'
-import {EOL} from 'os'
 
 import type {Extension} from '../main'
+import { ChkTeX } from './linterlib/chktex'
+import { LaCheck } from './linterlib/lacheck'
 
 export class Linter {
-    private readonly extension: Extension
+    private readonly chktex: ChkTeX
+    private readonly lacheck: LaCheck
     private linterTimeout?: NodeJS.Timer
-    private readonly currentProcesses = Object.create(null) as { [linterId: string]: ChildProcessWithoutNullStreams }
 
-    constructor(extension: Extension) {
-        this.extension = extension
+    constructor(private readonly extension: Extension) {
+        this.chktex = new ChkTeX(extension)
+        this.lacheck = new LaCheck(extension)
```

Comment:
I find it weird to explicitly instantiate these two classes before knowing if a linter will actually be used.